### PR TITLE
[Fix] Fix broken Webpack HMR when navigating to subroute returning 404

### PR DESCRIPTION
**Pull request by midori0507**
This fix live-reload on save

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -24,10 +26,23 @@ if(process.env.NODE_ENV !== 'development') {
...
+        stats: {
+            colors: true,
+            assets: false,
+            modules: false,
+        },
+    }));
     app.use(webpackHotMiddleware(compiler));
 }
 
+app.get('*', (req, res) => {
```
Comment (jimppa) [inline server/server.js]: This catch-all handler should be defined last in order, now requests to endpoints set by `addAuth()` never get executed.

File: config/webpack/prod.js [hunk 2]
```diff
@@ -90,7 +52,7 @@ const config = {
     plugins: [
         //new webpack.HotModuleReplacementPlugin(),
         new webpack.DefinePlugin({
-            'process.env.NODE_ENV': '"production"',
+            'process.env.NODE_ENV': 'production',
```
Comment (jimppa) [inline config/webpack/prod.js]: This breaks `yarn build/production`, should be as it was before (`'"production"'`).

**Comment by Rikuoja** [comment]
Okay, I'll have to defer this to @jussiarpalahti and @vikoivun as they are actually responsible for the current webpack config and deployment, don't know how this works with that.

File: config/webpack/prod.js (hunk 2 above)
Comment (midori0507) [reply config/webpack/prod.js]: Thats nice, ci passed and my build local worked. 
But anyway i’ll put it back, thought its a typo

**Pull request merged**